superset-frontend/src/views/CRUD/dashboard/DashboardList.tsx

```diff
@@ -444,6 +400,146 @@ class DashboardList extends React.PureComponent<Props, State> {
+    const handleExport = () => this.handleBulkDashboardExport([original]);
+    if (!this.canEdit && !this.canDelete && !this.canExport) {
+      return null;
+    }
+    return (
+      <span className="actions">
+        {this.canDelete && (
+          <ConfirmStatusChange
+            title={t('Please Confirm')}
+            description={
+              <>
+                {t('Are you sure you want to delete')}{' '}
+                <b>{original.dashboard_title}</b>?
+              </>
+            }
+            onConfirm={handleDelete}
+          >
+            {confirmDelete => (
+              <span
+                role="button"
+                tabIndex={0}
+                className="action-button"
+                onClick={confirmDelete}
+              >
+                <Icon name="trash" />
+              </span>
+            )}
+          </ConfirmStatusChange>
+        )}
+        {this.canExport && (
+          <span
+            role="button"
+            tabIndex={0}
+            className="action-button"
+            onClick={handleExport}
+          >
+            <Icon name="share" />
+          </span>
+        )}
+        {this.canEdit && (
+          <span
+            role="button"
+            tabIndex={0}
+            className="action-button"
+            onClick={handleEdit}
+          >
+            <Icon name="pencil" />
+          </span>
+        )}
+      </span>
+    );
+  }
+
+  renderCard = (props: Dashboard) => {
+    const menu = (
+      <Menu>
+        {this.canDelete && (
+          <Menu.Item>
+            <ConfirmStatusChange
+              title={t('Please Confirm')}
+              description={
+                <>
+                  {t('Are you sure you want to delete')}{' '}
+                  <b>{props.dashboard_title}</b>?
+                </>
+              }
+              onConfirm={() => this.handleDashboardDelete(props)}
+            >
+              {confirmDelete => (
+                <div
+                  role="button"
+                  tabIndex={0}
+                  className="action-button"
+                  onClick={confirmDelete}
+                >
+                  <ListViewCard.MenuIcon name="trash" /> Delete
+                </div>
+              )}
+            </ConfirmStatusChange>
+          </Menu.Item>
+        )}
+        {this.canExport && (
+          <Menu.Item
+            role="button"
+            tabIndex={0}
+            onClick={() => this.handleBulkDashboardExport([props])}
+          >
+            <ListViewCard.MenuIcon name="share" /> Export
+          </Menu.Item>
+        )}
+        {this.canEdit && (
+          <Menu.Item
+            role="button"
+            tabIndex={0}
+            onClick={() => this.openDashboardEditModal(props)}
+          >
+            <ListViewCard.MenuIcon name="pencil" /> Edit
+          </Menu.Item>
+        )}
+      </Menu>
+    );
+
+    return (
+      <ListViewCard
```

Comment:
With this file getting a little chunky, I'm starting to wonder if the Card, the table row, or any other atomic chunks of this file should get broken out into their own components.

Reply:
I think the chunkiness comes from these large blocks of JSX. There is a lot of markup in these files, however if we move them to their own component we'd have to pass in a lot of the context from this file somehow, things like permissions, handleDashboardDelete, handleDashboardExport, etc.

We could do something like `const createRenderDashboardCard = ({canDelete, canEdit, ..., handleDashboardDelete, ...}) => (props: Dashboard) => (<ListViewCard .....)` but that seems a little less readable, and adds another layer of abstraction, imho. 

I don't see any real benefit beyond reducing the number of lines of code in this component class. Similar results could be achieved using code folding features in the text editor. 

I do agree that there is a lot of shared logic between the various listviews, around permissions, data fetching, delete endpoints, etc. I'd like to consolidate some of this logic into a single `useListViewResource('dashboard')` hook, but that seems a little beyond the scope of this PR.

Reply:
Ok... might be worth filing a ticket for a rainy day.... all good for now.